src/Test/Perf/util/TraceManager_util.csx

```diff
@@ -0,0 +1,231 @@
+#load "ScenarioGenerator_util.csx"
+
+using System;
+using System.Diagnostics;
+using System.IO;
+
+interface ITraceManager
+{
+    int Iterations { get; }
+
+    void Cleanup();
+    void EndEvent();
+    void EndScenario();
+    void EndScenarios();
+    void ResetScenarioGenerator();
+    void Setup();
+    void Start();
+    void StartEvent();
+    void StartScenario(string scenarioName, string processName);
+    void Stop();
+    void WriteScenariosFileToDisk();
+}
+
+class TraceManagerFactory
+{
+    public static ITraceManager GetTraceManager(
+        int iterations = 3,
+        string cpcFolderPath = @"%SYSTEMDRIVE%\CPC",
+        string scenarioPath = @"%SYSTEMDRIVE%\CPC")
+    {
+        var cpcFullPath = Path.Combine(Environment.ExpandEnvironmentVariables(cpcFolderPath), "CPC.exe");
+        if (File.Exists(cpcFullPath))
+        {
+            var expandedScenarioPath = Environment.ExpandEnvironmentVariables(scenarioPath);
+            return new TraceManager(iterations, cpcFullPath, expandedScenarioPath);
+        }
+        else
+        {
+            return new NoOpTraceManager(iterations);
+        }
+    }
+}
+
+/// This is the Trace Manager that will be used when there is no CPC found in the machine
+/// All operations are NoOp
+class NoOpTraceManager : ITraceManager
```

Comment:
The WallClock timing is generated with the current flow. We can make this to calculate it but then the `csx` file will need to tell the TraceManager what the preparatory steps are and the what the test run is to compute the actual run time